api.js

```diff
@@ -14,73 +14,56 @@ function sendError(res, statusCode, code, description) {
 
 }
 
-app.get('/blobs/:blobId', (req, res) => {
-    db.query('select content_type, blob_value from blobs where blob_id = ?',
-        [req.params.blobId],
-        (err, data) => {
-            if (data.length != 1) {
-                sendError(res, 404, 'NotFound', 'Blob not found');
-            } else {
-                const row = data[0];
-                res.set('Content-Type', row.content_type);
-                res.send(row.blob_value);
-            }
-    });
+app.get('/blobs/:blobId', async (req, res) => {
+    const blobId = req.params.blobId;
+    const row = await db.oneOrNone(
+        'select content_type, blob_value from blobs where blob_id = $1',
+        [blobId]);
+    if (row === null) {
+        sendError(res, 404, 'NotFound', 'Blob not found');
+    } else {
+        res.set('Content-Type', row.content_type);
+        res.send(row.blob_value);
+    }
 });
 
-app.post('/contracts/:contractId/cfdi', (req, res) => {
+app.post('/contracts/:contractId/cfdi', async (req, res) => {
     if (!req.is("image/*") && !req.is("application/pdf")) {
         sendError(res, 415, "UnsupportedMediaType", "CFDI tiene formato inválido")
     }
     const contractId = req.params.contractId;
     const contractFile = req.body;
     const contentType = req.get('Content-Type');
-    db.query('select * from contracts where contract_id = ?',
-        [contractId],
-        (err, rows) => {
-            if (rows.length != 1) {
-                sendError(res, 404, 'NotFound', 'Contract not found');
-                return;
-            }
-            const contract = rows[0];
-            if (contract.cfdi_id !== null) {
-                sendError(res, 400, 'CFDIAlreadyExists', 'El contrato ya tiene CFDI');
-            } else {
-                db.query("insert into blobs (content_type, content) (?, ?)",
-                [contentType, contractFile],
-                (err, results) => {
-                    if (err) {
-                        sendError(res, 500, 'InternalServerError', 'Could not insert');
-                    } else {
-                        const blobId = results.insertId;
-                        db.query('update contracts where contract_id = ? set cfdi_id = ?',
-                            [contractId, blobId],
-                            (err) => {
-                                if (err) {
-                                    sendError(res, 500, 'InternalServerError', 'Could not update');
-                                } else {
-                                    res.status(204).end;
-                                }
-                            }
-                        )
-                    }
-                })
-            }
-        }
-    )

```

Comment:
Este ees un caso perfecto del pyramid of doom. Compara el codigo viejo con el nuevo.